src/instructlab/configuration.py

```diff
@@ -29,25 +31,63 @@
 # Local
 from . import log
 
+
+class STORAGE_DIR_NAMES:
+    ILAB = "instructlab"
+    DATASETS = "datasets"
+    CHECKPOINTS = "checkpoints"
+    MODELS = "models"
+    TAXONOMY = "taxonomy"
+    INTERNAL = (
+        "internal"  # for storing all ilab-internal files the user doesn't need to see
+    )
+
+
+def get_home_dir(*subdir: str) -> str:
+    return os.path.join(str(Path.home()), *subdir)
+
+
+def get_home_env_path(env: str, suffix: str, *home_paths: str) -> str:
+    """Neatly formats the environment paths for the user home & storage directories"""
+    home_path = os.getenv(env)
+    full_path = home_path if home_path else get_home_dir(*home_paths)
+    return os.path.join(full_path, suffix)
```

Comment:
Please don't roll your own platform directory code, use [platformdirs](https://pypi.org/project/platformdirs/). You are not pulling a new package. InstructLab has an indirect dependency on platformdirs already.

Reply:
@tiran I didn't know this already existed, thank you!

Reply:
Good idea, I'll replace all of this to use `platformdirs` instead.